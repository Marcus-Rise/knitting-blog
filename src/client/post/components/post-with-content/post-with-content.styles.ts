import styled from "styled-components";
import { BadScript, colors } from "../../../styles";

const ImageStyled = styled.div`
  height: 320px;

  width: auto;
  overflow: hidden;

  &:hover {
    cursor: pointer;
  }

  & > span {
    &,
    & > img {
      position: relative !important;
    }
  }

  img {
    object-fit: cover;
    height: 320px !important;
    max-height: none !important;
    min-height: 0 !important;
    margin: 0 !important;
  }
`;

const Title = styled.h2`
  ${BadScript};
  font-size: 2rem;
  color: ${colors.primary};
  text-align: center;
  text-transform: uppercase;
  margin: 0;
  line-height: initial;
`;

const Meta = styled.time`
  text-align: center;
  color: ${colors.secondary};
  text-transform: uppercase;
  display: block;
  margin: 1em 0;
`;

const Label = styled.p`
  ${BadScript};

  color: ${colors.primary};
  text-align: center;
  font-size: 1.5rem;
`;

const Footer = styled.time`
  color: ${colors.secondary};
  text-align: right;
  text-transform: capitalize;
  display: block;
  margin: 1em 0;
`;

export { ImageStyled, Title, Meta, Label, Footer };
