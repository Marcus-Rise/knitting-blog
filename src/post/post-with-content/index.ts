export { PostWithContent } from "./post-with-content.component";
