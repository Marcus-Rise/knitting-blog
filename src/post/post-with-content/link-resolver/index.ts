export { LinkResolver } from "./link-resolver.function";
