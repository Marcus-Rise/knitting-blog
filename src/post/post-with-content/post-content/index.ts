export { PostContent } from "./post-content.component";
