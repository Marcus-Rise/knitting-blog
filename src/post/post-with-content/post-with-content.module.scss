@use "src/styles/typography";

.image {
  $height: 320px;
  height: $height;

  width: auto;
  overflow: hidden;

  & > div {
    &, & > img {
      position: relative !important;
    }
  }

  img {
    object-fit: cover;
    height: $height !important;
    max-height: none !important;
    min-height: 0 !important;
    margin: 0 !important;
  }
}

.title {
  @include typography.BadScript;
  font-size: 2rem;
  color: var(--primary);
  text-align: center;
  text-transform: uppercase;
  margin: 0;
  line-height: initial;
}

.meta {
  text-align: center;
  color: var(--secondary);
  text-transform: uppercase;
}

.imageLabel {
  @include typography.BadScript;

  color: var(--primary);
  text-align: center;
  font-size: 1.5rem;
}

.description {
  line-height: 1.5rem;
  letter-spacing: 0.2em;
}

.footer {
  color: var(--secondary);
  text-align: right;
  text-transform: capitalize;
}
