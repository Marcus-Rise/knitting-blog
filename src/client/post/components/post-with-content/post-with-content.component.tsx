import type { FC } from "react";
import React from "react";
import { Container, Hr, ImageView } from "../../../components";
import Image from "next/image";
import { DateToString } from "../../../../common/utils/date-to-string";
import { PostContent } from "./post-content";
import type { IPost } from "../../../../common/post";
import { Footer, ImageStyled, Label, Meta, Title } from "./post-with-content.styles";

type IProps = IPost;

const PostWithContent: FC<IProps> = ({ content, date, imageLabel, imageSrc, title }) => {
  const dateStr = DateToString(date);

  return (
    <Container>
      <Title>{title}</Title>
      <Hr />
      <Meta>{dateStr}</Meta>
      {imageSrc && (
        <ImageView album={[{ src: imageSrc, alt: imageLabel }]}>
          <ImageStyled>
            <Image
              src={imageSrc}
              alt={imageLabel}
              layout={"fill"}
              priority
              placeholder={"blur"}
              blurDataURL={
                "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNcXw8AAeMBMA+N6mYAAAAASUVORK5CYII="
              }
            />
          </ImageStyled>
        </ImageView>
      )}
      <Label>{imageLabel}</Label>

      <PostContent content={content} />

      <Footer>{dateStr}</Footer>
    </Container>
  );
};

export { PostWithContent };
