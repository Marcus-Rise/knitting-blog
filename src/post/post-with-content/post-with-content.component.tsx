import React from "react";
import type { IPost } from "../post.interface";
import styles from "./post-with-content.module.scss";
import { Hr } from "../../components/hr";
import Image from "next/image";
import { DateToString } from "../../utils/date-to-string";
import { PostContent } from "./post-content";
import { ImageView } from "../../components/image-view";

type IProps = IPost;

const PostWithContent: React.FC<IProps> = (props) => {
  return (
    <>
      <h2 className={styles.title}>{props.title}</h2>
      <Hr />
      <p className={styles.meta}>{DateToString(props.date)}</p>
      {props.imageSrc && (
        <ImageView src={props.imageSrc} alt={props.imageLabel}>
          <div className={styles.image}>
            <Image
              src={props.imageSrc}
              alt={props.imageLabel}
              height={320}
              width={"auto"}
              layout={"responsive"}
              priority
            />
          </div>
        </ImageView>
      )}
      <p className={styles.imageLabel}>{props.imageLabel}</p>

      <PostContent content={props.content} />

      <p className={styles.footer}>{DateToString(props.date)}</p>
    </>
  );
};

export { PostWithContent };
