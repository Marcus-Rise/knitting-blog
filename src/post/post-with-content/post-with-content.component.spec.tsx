import { render } from "@testing-library/react";
import { PostWithContent } from "./post-with-content.component";
import React from "react";

jest.mock("../../utils/date-to-string", () => ({
  __esModule: true,
  DateToString: jest.fn((str) => str),
}));
jest.mock("../../components/image-view", () => ({
  __esModule: true,
  ImageView: jest.fn((props) => <div data-testid="image-view">{props.children}</div>),
}));
jest.mock("./post-content", () => ({
  __esModule: true,
  PostContent: jest.fn(() => <div data-testid={"content"} />),
}));

describe("PostWithContent", () => {
  test("render", () => {
    const { asFragment } = render(
      <PostWithContent
        title={"title"}
        slug={"slug"}
        date={"2020-11-11T17:47:51+0000"}
        imageSrc={"http://test.com"}
        imageLabel={"image"}
        description={"descr"}
        content={[]}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
