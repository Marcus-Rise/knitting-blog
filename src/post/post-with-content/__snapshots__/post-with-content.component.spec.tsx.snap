// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostWithContent render 1`] = `
<DocumentFragment>
  <h2
    class="title"
  >
    title
  </h2>
  <hr
    class="hr"
  />
  <p
    class="meta"
  >
    11 Nov, 2020
  </p>
  <div
    style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
  >
    <div
      style="display: block; box-sizing: border-box; padding-top: 100%;"
    />
    <img
      alt="image"
      decoding="async"
      src="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75"
      srcset="/_next/image?url=http%3A%2F%2Ftest.com&w=320&q=75 320w, /_next/image?url=http%3A%2F%2Ftest.com&w=420&q=75 420w, /_next/image?url=http%3A%2F%2Ftest.com&w=768&q=75 768w, /_next/image?url=http%3A%2F%2Ftest.com&w=1024&q=75 1024w, /_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75 1200w"
      style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
  </div>
  <p
    class="description"
  >
    descr
  </p>
  <ul>
    <li
      class="li"
    >
      BO – Bind off
    </li>
    <li
      class="li"
    >
      k2tog – knit two stitches together decrease
    </li>
    <li
      class="li"
    >
      p – purl
    </li>
    <li
      class="li"
    >
      YO – yarn over
    </li>
  </ul>
  <p
    class="description"
  >
    descr
  </p>
  <div
    class="row justify-content-center"
  >
    <div
      class="col-auto"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,<svg width=\\"NaN\\" height=\\"320\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
            style="max-width: 100%; display: block;"
          />
        </div>
        <img
          alt="image"
          decoding="async"
          src="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75"
          srcset="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75 1x"
          style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div
      class="col-auto"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,<svg width=\\"NaN\\" height=\\"320\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
            style="max-width: 100%; display: block;"
          />
        </div>
        <img
          alt="image"
          decoding="async"
          src="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75"
          srcset="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75 1x"
          style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div
      class="col-auto"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,<svg width=\\"NaN\\" height=\\"320\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
            style="max-width: 100%; display: block;"
          />
        </div>
        <img
          alt="image"
          decoding="async"
          src="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75"
          srcset="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75 1x"
          style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div
      class="col-auto"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,<svg width=\\"NaN\\" height=\\"320\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
            style="max-width: 100%; display: block;"
          />
        </div>
        <img
          alt="image"
          decoding="async"
          src="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75"
          srcset="/_next/image?url=http%3A%2F%2Ftest.com&w=1200&q=75 1x"
          style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
  </div>
  <hr
    class="hr"
    style="margin-top: 3rem; margin-bottom: 3rem;"
  />
  <p
    class="description"
  >
    descr
  </p>
  <p
    class="footer"
  >
    11 Nov, 2020
  </p>
</DocumentFragment>
`;
